torch/nn/modules/activation.py

```diff
@@ -756,30 +773,25 @@ def forward(self, query, key, value, key_padding_mask=None,
         - attn_output_weights: :math:`(N, L, S)` where N is the batch size,
           L is the target sequence length, S is the source sequence length.
         """
-        if hasattr(self, '_qkv_same_embed_dim') and self._qkv_same_embed_dim is False:
+        if not self._qkv_same_embed_dim:
             return F.multi_head_attention_forward(
                 query, key, value, self.embed_dim, self.num_heads,
-                None, self.in_proj_bias,  # set self.in_proj_weight = None
+                self.in_proj_weight, self.in_proj_bias,
```

Comment:
If `self._qkv_same_embed_dim` is False,  you don't have `self.in_proj_weight` here, right?

Reply:
The latest commit adds it as a `None` valued parameter